### PR TITLE
Allow to specify any host (not just github.com and githubenterprise.com)

### PR DESCRIPTION
**Pull request by kegesch**
Github Enterprise instances can have any domain.

### CODE REVIEW

File: src/GprTool/NuGetUtilities.cs [hunk 1]
```diff
@@ -41,8 +41,7 @@ public static bool BuildOwnerAndRepositoryFromUrl(PackageFile packageFile, strin
                 repositoryUrl = $"https://github.com/{repositoryUrl}";
             }
 
-            if (!Uri.TryCreate(repositoryUrl, UriKind.Absolute, out var repositoryUri) 
-                || !IsSupportedHost(repositoryUri.Host))
+            if (!Uri.TryCreate(repositoryUrl, UriKind.Absolute, out var repositoryUri))
```
Comment (N3rv0us) [inline src/GprTool/NuGetUtilities.cs:44]: You can also remove the private method `IsSupportedHost` and the definition of supported hosts `m_supportedHosts`

**Review by jcansdale — CHANGES_REQUESTED** [review]
Thanks for the PR!

Could you add some new test cases covering non-Gitub URLs here?
https://github.com/kegesch/gpr/blob/e1d27f1edee84439a029659f0057fcacf4ee4421/test/GprTool.Tests/NuGetUtilitiesTests.cs#L192

**Review by Qubitza — APPROVED** [review]

**Comment by kegesch** [comment]
@jcansdale I fixed the test case for `BuildOwnerAndRepositoryFromUrl` and added one where you suggested.
@Qubitza I removed the unused methods.

**Comment by jcansdale** [comment]
@kegesch it looks like I'll need to use my admin privileges to merge. Shall I go ahead and merge?

**Comment by kegesch** [comment]
@jcansdale you're right. Please go ahead.

**Comment by jcansdale** [comment]
@kegesch the new version has been published to DockerHub and nuget.org (it should appear in a few minutes)!
https://github.com/jcansdale/gpr/runs/7674852752?check_suite_focus=true#step:3:14

Thanks for your contribution! 😄

**Comment by kegesch** [comment]
Perfect! Thanks for the fast release 👍🏻

**Pull request merged**